lib/domain/client_provider/client_provider.dart

```diff
@@ -47,8 +48,31 @@ class ClientProviderImpl implements ClientProvider {
       onSuccess.call();
       _operationTypeRepository.reset();
     }).onError((error) {
-      debugPrint('Client initialization failed: ${error.runtimeType}.');
+      debugPrint('Client initialization failed: ${error.description}.');
+
+      // The SDK removes all the data when a rooted device or tampering
+      // is detected
+      if (Platform.isAndroid && _isNonRootedDeviceTamperingError(error)) {
+        // On Android the application must be closed when a generic device
+        // protection error is received (i.e. it is a checksum, debugger,
+        // emulator or instrumentation guard error)
+        exit(-1);
+      }
+
       _errorHandler.handle(error);
     }).execute();
   }
+
+  bool _isNonRootedDeviceTamperingError(
+    InitializationError initializationError,
+  ) {
+    // Return true if this is a tampering error different than rooted device
+    // error occurred (i.e. this is a checksum, debugger, emulator or
+    // instrumentation guard error).
+    return initializationError is InitializationDeviceProtectionError &&
+        initializationError is! InitializationRootedError &&
+        initializationError is! InitializationHardwareError &&
+        initializationError is! InitializationNoDeviceLockError &&
+        initializationError is! InitializationLockScreenHasChangedError;
+  }
```

Comment:
I would generally avoid "double negation", i.e. a method that returns a boolean with the true return value meaning that something is "not" something. Understanding such methods, especially as conditions is unnecessarily complex and could easily lead to mistakes.

Additionally, looking at the method body a true returns doesn't just mean that the error is not a rooted device tampering error, other type of errors are checked as well.

_For these reasons please add a different name to the method, and potentially flip the logic (although the safer option would be just changing the name) my recommendation would be, basing it on the code comment at the caller side `isGenericDeviceProtectionError`_

We quickly discussed this in-person and the method's name and logic originates from the Access App so I think this should be changed there as well so I'll create a ticket for that.

Reply:
Fixed.